### PR TITLE
ci: custom command wrapper for passing context and controlling output

### PR DESCRIPTION
**Pull request by Ajpantuso**
### Summary

Adds command wrapper to replace mage `sh.*` functions.

### Change Type

Docs/Test

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [x] All CI checks and tests are passing.

### CODE REVIEW

File: internal/tools/command.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// NewCommandAlias makes reuse of common commands possible by taking
+// a command name and variadic slice of command arguments and returing
```
Comment (sugarraysam) [inline internal/tools/command.go:14]: nit: "returning"

File: internal/tools/command.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		stdout bytes.Buffer
+		stderr bytes.Buffer
+	)
+
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if cfg.Verbose {
+		cmd.Stdout = io.MultiWriter(cmd.Stdout, os.Stdout)
+		cmd.Stderr = io.MultiWriter(cmd.Stderr, os.Stdout)
```
Comment (sugarraysam) [inline internal/tools/command.go:46]: cool pattern !

**Review by sugarraysam — APPROVED** [review]
Thanks for this. It significantly improves the readability of the `magefile.go`. I suppose you want to test this code a bit before merging it in our devkube repo?

LGTM

**Comment by Ajpantuso** [comment]
> Thanks for this. It significantly improves the readability of the `magefile.go`. I suppose you want to test this code a bit before merging it in our devkube repo?

I'm working on `Mage` for **AMO** right now so ideally:
1. This PR get merged
2. I make a second pass to clean things up/improve coverage
3. This moves to `devkube` or any shared repo
4. This is integrated with **AMO** mage file

I might push the **AMO** mage changes first just to get CI to run w/ `pre-commit` hopefully improving the `pr_check` time even more. Integrating can be a follow up.

**Pull request merged**